src/forms/AddUserForm.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FunctionComponent, useState } from "react";
+import { Form as FinalForm, Field } from "react-final-form";
+import { FORM_ERROR } from "final-form";
+import { TextField } from "../shared";
+import { Form, Button } from "reactstrap";
+import { getRequiredValidation } from "../shared/validators";
+import { useFormatMessage } from "../i18n/i18n.service";
+import axiosInstance from "../services/Axios";
+import { getCurrentUser } from "../actions/DataActions";
+import Notify from "../services/Notify";
+
+type AddUserFormTypes = {
+  value: {
+    username: string;
+    role: string;
+    password?: string;
+  };
+  afterSubmit: () => any;
+  onDismiss: () => any;
+};
+
+export const AddUserForm: FunctionComponent<AddUserFormTypes> = (props) => {
+  const [loading, setLoading] = useState(false);
+  const l10n = useFormatMessage();
+  const user = getCurrentUser();
+
+  const submitUser = async (body: any) => {
+    await axiosInstance.post(`/admin/identity/register?token=${user.token}`, body);
+  };
+
+  const onSubmit = async (formValue: any) => {
+    setLoading(true);
+    try {
+      await submitUser(formValue);
+      Notify.success(l10n("defaultSuccessMessage"));
+      props.afterSubmit();
+    } catch (e) {
+      Notify.error(e?.response?.data?.message ?? l10n("defaultErrorMessage"));
+
+      setLoading(false);
+      return { [FORM_ERROR]: "required.error" };
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const required = getRequiredValidation("required.error");
```

Comment:
Note: Because this is a stateless function call I recommend to move this outside function component body. Otherwise it gets created at every render.